### ISSUE TITLE
[MdOption] Cannot read property 'trim' of undefined

### ISSUE THREAD

**Issue by MNSolderstar** [vuematerial/vue-material#1863]
### Steps to reproduce

Any HTML in an md-option triggers a Vue Warn reporting an error.

```vue
 <md-field>
	<label for="profile">Process Profile</label>
	<md-select v-model="selected.profile" name="" :placeholder="placeholder.profile" id="profile">
		<md-option v-for="(packet,index) in list.profiles" :key="packet.id" :value="index">
			<span>Hello</span> World
		</md-option>
	</md-select>
</md-field>
```

### Which browser?
Google Chrome 67.0.3396.99 (Official Build) (64-bit)
VUE : **2.5.16**
VUE Material : **1.0.0-beta-10.2**

### What is actually happening?

> [Vue warn]: Error in created hook: "TypeError: Cannot read property 'trim' of undefined"
> TypeError: Cannot read property 'trim' of undefined

Triggered by source:

```js
getTextContent: function getTextContent() {
     if (this.$el) {
       return this.$el.textContent.trim();
     }

     var slot = this.$slots.default;

     return slot ? slot[0].text.trim() : ''; // <--- call to trim() on undefined in this line
},
```

### Reproduction Link

https://codesandbox.io/s/9o9jw9wy4y

**Comment by Samuell1** [comment]
Its because you using icon icon component inside.

**Comment by MNSolderstar** [comment]
Yes, I need to use color coded icons in the select list in my use case because a text label on it's own is insufficient but it does it if you put any html in there are all.
For example if you want to make the first word bold <strong>Hello</strong> World

**Comment by MNSolderstar** [comment]
I found a work around... putting &nbsp; in-front of the html tag suppresses the issue...

```
<md-option>
    &nbsp;<span>Hello</span> World
</md-option>
```

**Comment by Samuell1** [comment]
If you want to use icons in options use Menu component, select is not created to handle other values then text inside.

**Comment by MNSolderstar** [comment]
Yes, but if the selection event got the selected items label **html** content **instead of** trying to get **text** only content to use in the select menus selected item display then it would work fine.

There is very little stopping this component from supporting html labels it would be counter intuitive to not have a html select menu support html labeling, after all this does not effect the value set in the :value attribute.

**Comment by okolomiets** [comment]
any updates on this issue/proposal? it's still not working as expected.

**Comment by JuergenSimon** [comment]
As of August 2020 there is still no movement on this. Ran into this knife in the belly as well.

**Comment by saylestyler** [comment]
> As of August 2020 there is still no movement on this. Ran into this knife in the belly as well.

extremely knife in the belly 😭

**Comment by abhimanusharma** [comment]
> As of August 2020 there is still no movement on this. Ran into this knife in the belly as well.
@Samuell1 
Extremely knife in the belly. Still no update.

**Comment by isaachvazquez** [comment]
> As of August 2020 there is still no movement on this. Ran into this knife in the belly as well.

Still no update. Had to duplicate the `md-option` component into our app, and rewrite it to work for our needs.
